dyno-core/src/test/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallbackTest.java

```diff
@@ -326,6 +326,37 @@ public void testGetConnectionsFromRingWhenHostDown() throws Exception {
 		verifyExactly(hostnames, "h5", "h6");
 	}
 
+    @Test
+    public void testReplicationFactorOf3WithDupes() {
```

Comment:
There is a test for RF2, [here](https://github.com/jcacciatore/dyno/blob/3b482d9d0ed41a4ccd6bb595dc79edc779686d21/dyno-core/src/test/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallbackTest.java#L382).

The client contacts every dynomite server in determining the topology therefore it sees a lot of duplicate info.